tests/compas/files/test_xml.py

```diff
@@ -49,3 +49,12 @@ def test_namespaces_to_string():
     # Note: Minidom does some funny things to namespaces.  First, if a namespace isn't used, it will be stripped out.
     # Second, it will include the original namespace declaration, but also repeat that declaration with another name,
     # and replace all references to the original with the new.
+
+def test_default_namespace_to_string():
+    xml = XML.from_string(
+        """<?xml version="1.0" encoding="UTF-8"?><robot xmlns="https://default.org/namespace" xmlns:xacro="http://www.ros.org/wiki/xacro" name="panda"><xacro:bamboo/></robot>"""
+    )
+    xml_string = xml.to_string(prettify=True)
```

Comment:
Interesting that this test passes! That means we have another issue somewhere else, because my `acrobot.urdf` test still fails. The main difference being that I don't test full round trip, but just parse and try to access values:

```python
@pytest.fixture
def acrobot_file():
    return os.path.join(BASE_FOLDER, 'fixtures', 'acrobot.urdf')

def test_acrobot_urdf(acrobot_file):
    r = RobotModel.from_urdf_file(acrobot_file)
    assert r.name == 'Acrobot'

```

Reply:
🤔

Reply:
Oh, dear, something is very broken indeed.  Your `r` isn't even a `RobotModel` instance.

Reply:
When there is a default namespace, all of the tags are prepended with `{<default xmlns value>}` in the python xml element tree parser. So when creating the `URDF`, `parse_element` is passed `{https://drake.mit.edu}robot` as a path rather than simply `robot`.  (This behavior exists on the main branch currently.). Since the `URDFParser` isn't registered with such a path, it creates a `URDFGenericElement` instead of a `RobotModel`.  There is a way to interrupt the parse and inject our own tags stripped of the namespace, but I don't think that's wise.  Maybe a drake robot is indeed a different creature than an unmodified robot.  What do you think?

Reply:
Yep, it is a pretty crappy situation. The only "solution" I found on the interwebz was doing a search/replace to strip that empty default namespace from the text. I tested it on the `XmlReader.from_string()` and it works, but I did not find a way to do it for the `XmlReader.from_file()`. A first naive approach was to read all file content and treat it as string, but that failed miserably when feeding it the `test_model.py` examples that does in-memory io from a zip file.

We could leave it as it is (since it is already like this on master), however, I don't know when it broke because it used to work and it should be fixed at some point. I suspect there's likely a way to hijack the XML parser and ignore the default  namespace.

Reply:
I've been digging around and not found anything acceptable.  So should I make an issue of this and merge this PR as is?

Reply:
Yes, as discussed, this is good to go, especially now that you added the extra (marked as skip) test for this situation.